src/VirtualClient/VirtualClient.Actions.UnitTests/Wrathmark/WrathmarkTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace VirtualClient.Actions.Wrathmark.UnitTests
+{
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using NUnit.Framework;
+    using VirtualClient.Actions.Wrathmark;
+    using VirtualClient.Contracts;
+
+    [TestFixture]
+    [Category("Unit")]
+    public class WrathmarkResultParserTests
+    {
+        private static readonly string ProfilesDirectory = Path.Combine(
+            Path.GetDirectoryName(Assembly.GetAssembly(typeof(WrathmarkResultParserTests)).Location),
+            "Examples",
+            "Wrathmark");
+
+        [Test]
+        public void Parse_Benchmark_SingleResult_ReturnsMetrics()
+        {
+            // Arrange
+            string results = File.ReadAllText(Path.Combine(ProfilesDirectory, "BenchWrathmark.txt"));
+            WrathmarkMetricsParser sut = new WrathmarkMetricsParser(results);
+
+            // Act
+            var metrics = sut.Parse();
+
+            // Assert
+            Assert.IsNotNull(metrics);
+            Assert.AreNotEqual(0, metrics.Count);
+            Assert.AreEqual(1, metrics.Count);
+            Assert.That(metrics.All(m => m.Name != null));
+            Assert.That(metrics.All(m => m.Value >= 0));
+
+            Metric expected = new Metric("BoardsPerSecond", 7_760_880, MetricRelativity.HigherIsBetter);
+            Metric actual = metrics[0];
+
+            Assert.AreEqual(expected.Name, actual.Name);
```

Comment:
If you would like, you could use our extension methods like
MetricAssert.Exists(this.actualMetrics, "device_peak_memory", 3113.97, "MB");
They have good error message so that it tells you whether it's name, or value or unit is wrong